autopilot/prefattach.go

```diff
@@ -72,12 +77,63 @@ func (p *PrefAttachment) NodeScores(g ChannelGraph, chans []Channel,
 	chanSize btcutil.Amount, nodes map[NodeID]struct{}) (
 	map[NodeID]*NodeScore, error) {
 
-	// Count the number of channels for each particular node in the graph.
+	// We first run though the graph once in order to find the median
+	// channel size.
+	var (
+		allChans       []btcutil.Amount
+		seenChans      = make(map[uint64]struct{})
+		medianChanSize btcutil.Amount
+	)
+	if err := g.ForEachNode(func(n Node) error {
+		err := n.ForEachChannel(func(e ChannelEdge) error {
+			if _, ok := seenChans[e.ChanID.ToUint64()]; ok {
+				return nil
+			}
+			seenChans[e.ChanID.ToUint64()] = struct{}{}
+			allChans = append(allChans, e.Capacity)
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	sort.Slice(allChans, func(i, j int) bool {
```

Comment:
Duplicate code to what was added in the server, we can add an abstract method in the `router` package perhaps? Or a "static method" (lol) on the `ChannelGraph`

Reply:
added `autopilot.Median()`